src/server/scripts/Outland/TempestKeep/Eye/boss_void_reaver.cpp

```diff
@@ -29,107 +30,89 @@ enum voidReaver
 };
 
-class boss_void_reaver : public CreatureScript
+struct boss_void_reaver : public BossAI
 {
-public:
-    boss_void_reaver() : CreatureScript("boss_void_reaver") { }
-
-    struct boss_void_reaverAI : public BossAI
+    boss_void_reaver(Creature* creature) : BossAI(creature, DATA_REAVER)
     {
-        boss_void_reaverAI(Creature* creature) : BossAI(creature, DATA_REAVER)
-        {
-            me->ApplySpellImmune(0, IMMUNITY_DISPEL, DISPEL_POISON, true);
-            me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_HEALTH_LEECH, true);
-            me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_POWER_DRAIN, true);
-            me->ApplySpellImmune(0, IMMUNITY_STATE, SPELL_AURA_PERIODIC_LEECH, true);
-            me->ApplySpellImmune(0, IMMUNITY_STATE, SPELL_AURA_PERIODIC_MANA_LEECH, true);
-        }
-
-        void Reset() override
+        scheduler.SetValidator([this]
         {
-            BossAI::Reset();
-        }
+            return !me->HasUnitState(UNIT_STATE_CASTING);
+        });
+
+        me->ApplySpellImmune(0, IMMUNITY_DISPEL, DISPEL_POISON, true);
+        me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_HEALTH_LEECH, true);
+        me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_POWER_DRAIN, true);
+        me->ApplySpellImmune(0, IMMUNITY_STATE, SPELL_AURA_PERIODIC_LEECH, true);
+        me->ApplySpellImmune(0, IMMUNITY_STATE, SPELL_AURA_PERIODIC_MANA_LEECH, true);
+    }
 
-        void KilledUnit(Unit* victim) override
-        {
-            if (victim->GetTypeId() == TYPEID_PLAYER && roll_chance_i(50))
-                Talk(SAY_SLAY);
-        }
+    void Reset() override
+    {
+        BossAI::Reset();
+        _recentlySpoken = false;
+    }
 
-        void JustDied(Unit* killer) override
+    void KilledUnit(Unit* victim) override
+    {
+        if (!_recentlySpoken)
         {
-            Talk(SAY_DEATH);
-            BossAI::JustDied(killer);
+            Talk(SAY_SLAY);
+            _recentlySpoken = true;
+            scheduler.Schedule(5s, [this](TaskContext)
+            {
+                _recentlySpoken = false;
+            });
         }
+    }
 
-        void JustEngagedWith(Unit* who) override
-        {
-            Talk(SAY_AGGRO);
-            BossAI::JustEngagedWith(who);
+    void JustDied(Unit* killer) override
+    {
+        Talk(SAY_DEATH);
+        BossAI::JustDied(killer);
+    }
 
-            events.ScheduleEvent(EVENT_SPELL_POUNDING, 15000);
-            events.ScheduleEvent(EVENT_SPELL_ARCANEORB, 3000);
-            events.ScheduleEvent(EVENT_SPELL_KNOCK_AWAY, 30000);
-            events.ScheduleEvent(EVENT_SPELL_BERSERK, 600000);
-            me->CallForHelp(105.0f);
-        }
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        Talk(SAY_AGGRO);
+        me->CallForHelp(105.0f);
 
-        void UpdateAI(uint32 diff) override
+        scheduler.Schedule(600s, [this](TaskContext context)
```

Comment:
```suggestion
        scheduler.Schedule(10min, [this](TaskContext context)
```

very nitpicky but looks nicer